pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -339,16 +354,16 @@ func (a *Actuator) CreateMachine(cluster *clusterv1.Cluster, machine *clusterv1.
 		return nil, err
 	}
 	// Add tags to the created machine
-	tagList := []*ec2.Tag{}
+	rawTagList := []*ec2.Tag{}
 	for _, tag := range machineProviderConfig.Tags {
-		tagList = append(tagList, &ec2.Tag{Key: aws.String(tag.Name), Value: aws.String(tag.Value)})
+		rawTagList = append(rawTagList, &ec2.Tag{Key: aws.String(tag.Name), Value: aws.String(tag.Value)})
 	}
-	tagList = append(tagList, []*ec2.Tag{
+	rawTagList = append(rawTagList, []*ec2.Tag{
```

Comment:
Can't we just put all the tags under a map and then check if the key already exists?
```go
tagMap := make(map[string]struct{})
for _, tag := range machineProviderConfig.Tags {
 		tagMap[tag.Name] = struct{}{}
}
if _, exists := tagMap["clusterid"]; !exists {
 		tagList = append(tagList, {Key: aws.String("clusterid"), Value: aws.String(clusterID)}}
}
...
```

Reply:
personally I like current implementation. It prevents also from injecting duplications  @frobware wdyt?